blaze-core/src/main/scala/org/http4s/blazecore/util/BodylessWriter.scala

```diff
@@ -13,33 +13,32 @@ import fs2.Stream._
 import fs2.interop.cats._
 import fs2.util.Attempt
 import org.http4s.blaze.pipeline._
+import org.http4s.util.StringWriter
 
 /** Discards the body, killing it so as to clean up resources
   *
   * @param headers ByteBuffer representation of [[Headers]] to send
   * @param pipe the blaze `TailStage`, which takes ByteBuffers which will send the data downstream
   * @param ec an ExecutionContext which will be used to complete operations
   */
-class BodylessWriter(headers: ByteBuffer, pipe: TailStage[ByteBuffer], close: Boolean)
-                    (implicit protected val ec: ExecutionContext) extends EntityBodyWriter {
+class BodylessWriter(pipe: TailStage[ByteBuffer], close: Boolean)
+  (implicit protected val ec: ExecutionContext) extends Http1Writer {
+  private[this] var headers: ByteBuffer = null
 
-  private lazy val doneFuture = Future.successful( () )
+  def writeHeader(headerWriter: StringWriter): Future[Unit] =
+    pipe.channelWrite(Http1Writer.headersToByteBuffer(headerWriter.result))
 
   /** Doesn't write the entity body, just the headers. Kills the stream, if an error if necessary
     *
     * @param p an entity body that will be killed
     * @return the Task which, when run, will send the headers and kill the entity body
     */
-  override def writeEntityBody(p: EntityBody): Task[Boolean] = Task.async { cb =>
-    val callback = cb.compose((t: Attempt[Unit]) => t.map(_ => close))
+  override def writeEntityBody(p: EntityBody): Task[Boolean] =
+    p.drain.run.map(_ => close)
```

Comment:
I think thisi s a safe refactoring, with the work now done in `writeHeader`, but a second set of brains would be appreciated here.